libvast/src/system/node.cpp

```diff
@@ -613,15 +613,35 @@ node_state::spawn_command(const invocation& inv,
   return spawn_actually(args);
 }
 
-node_actor::behavior_type node(node_actor::stateful_pointer<node_state> self,
-                               std::string name, std::filesystem::path dir) {
+node_actor::behavior_type
+node(node_actor::stateful_pointer<node_state> self, std::string name,
+     std::filesystem::path dir, bool accounting) {
   self->state.name = std::move(name);
   self->state.dir = std::move(dir);
   // Initialize component and command factories.
   node_state::component_factory = make_component_factory();
   node_state::command_factory = make_command_factory();
+  // Initialize the accountant.
+  auto accountant = accountant_actor{};
+  if (accounting) {
+    auto metrics_opts = caf::get_or(content(self->system().config()),
+                                    "vast.metrics", caf::settings{});
+    auto cfg = to_accountant_config(metrics_opts);
+    if (!cfg) {
+      VAST_ERROR("{} failed to parse metrics configuration: {}", *self,
+                 cfg.error());
+    } else {
+      accountant = self->spawn(vast::system::accountant, std::move(*cfg),
+                               self->state.dir);
+      auto err = register_component(
+        self, caf::actor_cast<caf::actor>(accountant), "accountant");
+      VAST_ASSERT(err
+                  == caf::none); // Registration cannot fail; empty registry.
+    }
+  }
   // Initialize the file system with the node directory as root.
-  auto fs = self->spawn<caf::detached>(posix_filesystem, self->state.dir);
+  auto fs
+    = self->spawn<caf::detached>(posix_filesystem, self->state.dir, accountant);
```

Comment:
accountant can be moved, but it's not a big deal i guess

Reply:
Good spot. I made that change while I moved the code over.